### PR TITLE
Copy a memory address formatted as a raw data. (edit)

### PR DESCRIPTION
**Pull request by Rat431**
Implement a memory address copying in the clipboard as a raw-data format from the x64dbg's context menu.

### CODE REVIEW

File: src/gui/Src/Gui/CPUDisassembly.cpp [hunk 1]
```diff
@@ -288,11 +288,13 @@ void CPUDisassembly::setupRightClickContextMenu()
     mMenuBuilder->addMenu(makeMenu(DIcon("binary.png"), tr("&Binary")), binaryMenu);
 
     MenuBuilder* copyMenu = new MenuBuilder(this);
+    QAction* rawAddrAction = makeAction(DIcon("copy_address.png"), tr("Address to pattern"), SLOT(copyRawAddressSlot()));
```
Comment (mrexodia) [inline src/gui/Src/Gui/CPUDisassembly.cpp:291]: You can directly inline this in the `addAction` below

File: src/gui/Src/Gui/CPUDisassembly.cpp [hunk 2]
```diff
@@ -1556,6 +1558,25 @@ void CPUDisassembly::copyAddressSlot()
     Bridge::CopyToClipboard(clipboard);
 }
 
+void CPUDisassembly::copyRawAddressSlot()
```
Comment (mrexodia) [inline src/gui/Src/Gui/CPUDisassembly.cpp:1561]: What exactly does this copy? The byte pattern `AA BB CC DD` or am I misunderstanding?

image

If I understand correctly this should do the same.

**Review by mrexodia — CHANGES_REQUESTED** [review]
Thanks for the contribution. I didn't have time to try it (on the run), but added some comments for clarification.

File: src/gui/Src/Gui/CPUDisassembly.cpp (hunk 2 above)
Comment (Rat431) [reply src/gui/Src/Gui/CPUDisassembly.cpp:1561]: Hi, go in the "Copy" section and look for "Address to pattern". This option will just take the memory address formatted as a raw pattern, ready to be written (pasted) in memory resulting as a valid pointer without having to manually decompose the address.

**Comment by Rat431** [comment]
> ✅ [Download x64dbg 1.0.1336](https://ci.appveyor.com/project/mrexodia/x64dbg/builds/43465432/artifacts) (commit [f69c3932aa](https://github.com/x64dbg/x64dbg/commit/f69c3932aa) by @Rat431)

Just made the change you requested.

**Comment by mrexodia** [comment]
Thanks for your contribution, but I don't think this use case is frequent enough to warrant a new menu item. It should be pretty straightforward to do this with a plugin or you can do something like `bswap(dis.sel())`. It could be great to add a `Bytes:` column to the calculation like the register edit dialog though because I use the `bswap()` function quite a bit when using x64dbg as a calculator.

**Comment by mrexodia** [comment]
I implemented this feature in the calculator dialog, thanks for the idea!

image

**Pull request closed without merging**